onyo/lib/onyo.py

```diff
@@ -1145,54 +1145,53 @@ def _read_asset(self, asset: Path) -> dict:
     def _set_sanitize(self, paths: Iterable[Union[Path, str]],
                       depth: Union[int, None]) -> list[Path]:
         """
-        Check and normalize a list of paths. Traverse directories and select all
-        assets from `paths` and return them as a list of absolute Paths.
+        Check and normalize a list of paths. Select all assets in the
+        repository that are relative to the given `paths` descending at most
+        `depth` directories. A `depth` of 0 descends without a limit.
         """
-        error_path_absent = []
-        error_path_protected = []
-        paths_to_set = []
-
         if depth and depth < 0:
             log.error(f"depth values must be positive, but is {depth}.")
             raise ValueError(f"depth values must be positive, but is {depth}.")
 
-        for p in paths:
-            full_path = Path(self.opdir, p).resolve()
-
-            # paths must exist
-            if not full_path.exists():
-                error_path_absent.append(p)
-                continue
-
-            # protected paths
-            if self._is_protected_path(full_path):
-                error_path_protected.append(p)
-                continue
-
-            if full_path.is_dir():
-                paths_to_set += self._select_assets_from_directory(full_path, depth)
-            else:
-                paths_to_set.append(full_path)
+        paths = {Path(p) for p in paths}
+
+        # TODO: This is superfluous due to using self.assets to find valid
```

Comment:
In my opinion this type of user input should be validated on the CLI side, rather than in the API because the purpose is to inform the user they have entered a non-existent or protected path. On the API side this doesn't actually do any harm, as there will never be assets to set or unset in a protected or non-existing path. What are we protecting API users from with this validation check, other than restricting their input?

Reply:
Hmmm. My initial response is: why treat CLI consumers differently from API consumers?

What you're proposing has much bigger implications, which is fine, but it would make it easier to review if you presented it as a change in Onyo's behavior (which would need to be changed  everywhere).

If I understand you correctly, you're proposing that Onyo accept input that is effectively no-op (aka, trying to `onyo set` under `.git/`. And since it won't do anything (since there are no assets there), then the user should not be informed.

Following that logic, I'm not sure why they should be informed if the target paths don't exist, etc. At which point, how is the user to know if what they provided as input is not-allowed/ignored by Onyo?